tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTests.cs

```diff
@@ -219,5 +254,25 @@ public void TestTextIndexCreation()
             Assert.Equal("spanish", index["default_language"].AsString);
             Assert.Equal(1, index["key"]["c"].AsInt32);
         }
+
+        [Fact]
+        public void Wildcard_should_return_expected_result()
+        {
+            var keys = IndexKeys.Wildcard();
+            BsonDocument expected = BsonDocument.Parse("{ \"$**\" : 1 }");
+            keys.ToBsonDocument().Should().Be(expected);
+
+            keys = IndexKeys.Wildcard("b");
+            expected = BsonDocument.Parse("{ \"b.$**\" : 1 }");
+            keys.ToBsonDocument().Should().Be(expected);
+
+            keys = IndexKeys.Wildcard("$**");
+            expected = BsonDocument.Parse("{ \"$**\" : 1 }");
+            keys.ToBsonDocument().Should().Be(expected);
+
+            keys = IndexKeys.Wildcard("b.$**");
```

Comment:
Should this even be allowed? Or should we require "b" and add the ".$**" automatically?

What is the purpose of providing two ways to do the same thing. More than one way to do the same thing can cause confusion.

Reply:
Answered above.